firefox compatibility adjustments

RobertFridolin opened the pull request:
fix: small adjustments for firefox compatibility

#5

RobertFridolin committed:
firefox compatibility adjustments

small adjustments for firefox compatibility

RobertFridolin commented:
these are only preparations to hande different browsers that should otherwise have no consequences 
no problem-free Firefox compatibility yet

ZoeBijl commented on extension/js/compatibility.js:
Can you explain what this does?

RobertFridolin replied:
maps firefox specific API to chrome api naming for example browser.storage.local.get to chrome.storage.local.get

ZoeBijl commented on extension/modules/about-dialog.js:
```diff
@@ -76,6 +76,7 @@ class AboutDialog {
             <img src="${chrome.runtime.getURL('images/AES-logo-128.png')}">
             <h2>AirlineSim Enhancement Suite</h2>
             <p>Version ${version}</p>
+            <p>Browser ${window.navigator.userAgent}<p>
```
What’s your motivation for putting the userAgent string in the AES about screen?

RobertFridolin replied:
was for debug purposes if someone sends you a bug report on discord you have the AES version and exact browser version right there, in case there are browser-specific problems
if it bothers you, it can also be removed/shortened or moved elsewhere or hidden behind a "debug information" button or something

RobertFridolin commented:
was for debug purposes if someone sends you a bug report on discord you have the AES version and exact browser version right there, in case there are browser-specific problems
if it bothers you, it can also be removed or moved elsewhere or hiden behind a "debug information" button or something

RobertFridolin commented:
if (typeof chrome === 'undefined') {
    var chrome = browser;
}
maps firefox specific API to chrome api naming like browser.storage.local.get to chrome.storage.local.get

changes in extension/modules/aes-menu.js made because of no match is found on firefox then an error occurs there
I would also prefer the full userAgent output to the simple versioning, as it also contains information about the platform, browser derivatives or similar.

RobertFridolin committed:
firefox compatibility adjustments

make site load detection more robust by checking
for readystate=complete because the load event might get missed

fix data saving by not saving methods in database, makes no sense doing so?

#5

RobertFridolin committed:
Merge branch 'ZoeBijl:next' into firefox-compatibility

RobertFridolin committed:
update errorhandling

RobertFridolin committed:
Merge branch 'firefox-compatibility' of https://github.com/RobertFridolin/airlinesim-enhancement-suite into firefox-compatibility
